torch/csrc/api/src/serialize/input-archive.cpp

```diff
@@ -41,7 +41,7 @@ void InputArchive::read(
   if (tensor.defined()) {
     torch::NoGradGuard guard;
     if (tensor.device() != read_tensor.device()) {
-      tensor.set_data(autograd::Variable(read_tensor).data());
+      tensor.set_data(read_tensor);
```

Comment:
This is used for changing the device of the returned `tensor` (e.g. from CPU to CUDA) when reading tensor data from a serialized file. The reason we need this branch is that the else-branch `tensor.set_(read_tensor)` cannot achieve the same goal, since `tensor.set_(read_tensor)` internally calls `checked_tensor_unwrap()` and enforces that `tensor` and `read_tensor` have the same device type. If we only do `tensor.set_(read_tensor)`, the test in https://github.com/pytorch/pytorch/blob/bc5398451e5de266767120a18f75e08f06d4bf8e/test/cpp/api/serialize.cpp#L265-L271 will fail.